### PR TITLE
exitStatus -> commandExitResult

### LINKED ISSUE #167582: Terminal quick fixes: exitStatus flag is confusing
Testing #166762

The behavior I see is that with `"exitStatus": true`, the provider only runs when the exit code was 0. With `"exitStatus": false`, it only runs when non-0. Is that right? I think there should be an option for my provider to run regardless of exit code.

### PR DESCRIPTION
**Pull request by meganrogge**
fix https://github.com/microsoft/vscode/issues/167582
fyi @connor4312 to update the npm contrib

### CODE REVIEW

**Comment by connor4312** [comment]
When renaming or making API changes, it is probably good to also update the built-in extension; useful to test the change and also prevent the possibility of a broken release if one goes out between the time that a breaking change is merged and I have an opportunity to create, submit, and merge an additional PR.

**Comment by Tyriar** [comment]
Does the .d.ts not get used? tsc would be able to ensure we don't break the extensions if so

**Comment by meganrogge** [comment]
@Tyriar it does not because the command selector is a contribution

**Comment by connor4312** [comment]
The .d.ts does get used, but changes to contributions are not validated in that.

> I was also wondering if there are tests we could add that would fail if I were to forget to update the builtin extension?

Hmm, we'd need to do some browser integration test type thing for that, since there isn't an API to check if a light bulb is shown in the UI

File: src/vs/workbench/contrib/terminal/common/terminal.ts [hunk 1]
```diff
@@ -781,9 +781,10 @@ export const terminalContributionsDescriptor: IExtensionPointDescriptor<ITermina
 								}
 							}
 						},
-						exitStatus: {
-							description: nls.localize('vscode.extension.contributes.terminal.quickFixes.exitStatus', "True if the exit code is non-zero"),
-							type: 'boolean',
+						commandExitResult: {
+							description: nls.localize('vscode.extension.contributes.terminal.quickFixes.commandExitResult', "The command result to match on"),
+							enum: ['success', 'error'],
+							enumDescriptions: ['zero exit code', 'non-zero exit code']
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/common/terminal.ts:787]:
```suggestion
							enumDescriptions: [
								'The command exited with an exit code of zero.',
								'The command exited with a non-zero exit code.'
							]
```

**Review by Tyriar — CHANGES_REQUESTED** [review]

**Review by Tyriar — APPROVED** [review]

**Pull request merged**